src/tools/rustfmt/src/types.rs

```diff
@@ -1093,6 +1093,19 @@ impl Rewrite for ast::TyPat {
             ast::TyPatKind::Range(ref lhs, ref rhs, ref end_kind) => {
                 rewrite_range_pat(context, shape, lhs, rhs, end_kind, self.span)
             }
+            ast::TyPatKind::Or(ref variants) => {
+                let mut first = true;
+                let mut s = String::new();
+                for variant in variants {
+                    if first {
+                        first = false
+                    } else {
+                        s.push_str(" | ");
+                    }
+                    s.push_str(&variant.rewrite_result(context, shape)?);
+                }
+                Ok(s)
+            }
```

Comment:
Do we have a syntax for this? looking at the UI tests I'm just seeing macro calls.

Also, I'm not sure the formatting implementation is correct. The `TyPatKind::Or` could exceed the `max_width` without wrapping if it's very long. I feel like we should implement this similarly to [`PatKind::Or`](https://github.com/rust-lang/rustfmt/blob/fd0ea742f849e2ff270b586f0a28b7c1e69aa8aa/src/patterns.rs#L103-L129).

If possible, can you add a test case with the relevant `#![feature()]` annotation?

Reply:
The only syntax is the pattern type macro

Reply:
Added to https://github.com/rust-lang/rust/issues/123646 as we need to figure out syntax before stabilization, but that's not relevant for https://github.com/rust-lang/rust/issues/135996